hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/common/model/HoodieSparkRecord.java

```diff
@@ -384,28 +387,35 @@ private static boolean hasMetaFields(StructType structType) {
   private static HoodieRecord<InternalRow> convertToHoodieSparkRecord(StructType structType, HoodieSparkRecord record, boolean withOperationField) {
     return convertToHoodieSparkRecord(structType, record,
         Pair.of(HoodieRecord.RECORD_KEY_METADATA_FIELD, HoodieRecord.PARTITION_PATH_METADATA_FIELD),
-        withOperationField, Option.empty());
+        withOperationField, Option.empty(), Option.empty());
   }
 
   private static HoodieRecord<InternalRow> convertToHoodieSparkRecord(StructType structType, HoodieSparkRecord record, boolean withOperationField,
-      Option<String> partitionName) {
+      Option<String> partitionName, Option<StructType> structTypeWithoutMetaFields) {
     return convertToHoodieSparkRecord(structType, record,
         Pair.of(HoodieRecord.RECORD_KEY_METADATA_FIELD, HoodieRecord.PARTITION_PATH_METADATA_FIELD),
-        withOperationField, partitionName);
+        withOperationField, partitionName, structTypeWithoutMetaFields);
   }
 
   /**
    * Utility method to convert bytes to HoodieRecord using schema and payload class.
    */
   private static HoodieRecord<InternalRow> convertToHoodieSparkRecord(StructType structType, HoodieSparkRecord record, Pair<String, String> recordKeyPartitionPathFieldPair,
-      boolean withOperationField, Option<String> partitionName) {
+      boolean withOperationField, Option<String> partitionName, Option<StructType> structTypeWithoutMetaFields) {
```

Comment:
can you enhance the java doc on when and how to use this method. for eg, when should the last arg be set? 
or should we introduce overloaded method.